test/controllers/users_controller_test.rb

```diff
@@ -402,15 +402,12 @@ class UsersControllerTest < ActionController::TestCase
     assert_redirected_to ApplicationHelper.current_hosts_path
     assert_match /mail.*invalid/i, flash[:warning]
 
-    # Subsequent redirects to the user edit page should preserve the warning
+    # User is able to log in without the email
     user = User.unscoped.find_by_login('ldap-user')
+    # Adding viewer role so the user is able to access the default page (hosts), otherwise the response is 403
+    user.roles << roles(:viewer)
     get :index, session: set_session_user.merge(:user => user.id)
-    assert_redirected_to edit_user_path(user)
-
-    get :edit, params: { :id => user.id }, session: set_session_user.merge(:user => user.id)
     assert_response :success
-    assert_match /mail.*invalid/i, flash[:warning]
-    assert_match /An email address is required/, flash[:error]
```

Comment:
Why did you choose to remove it instead of testing it with `mail_enabled set to true?

Reply:
I was so focused on external user being able to log in without email (the initial goal of this PR) that this did not cross my mind :smile:. I will fix it.

Reply:
Fixed.